### PR TITLE
Refactor AdaPool, BatchToSpace, and BatchNorm shared tests

### PR DESCRIPTION
**Pull request by olpipi**
### Details:
- Refactor AdaPool, BatchToSpace, and BatchNorm layer tests to use SubgraphBaseTest
- Update cpu plugin instantiation of these tests to use new implementation
- Gtest requires to have test fixture class name which is not equal to old class. That's why new tests are placed to new directory. This way we can use the same names and don't mix old and new tests together in Gtest

### Tickets:
 - [CVS-111364](https://jira.devtools.intel.com/browse/CVS-111364)

### CODE REVIEW

File: src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/adaptive_pooling.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+#include "shared_test_classes/base/ov_subgraph.hpp"
+
+namespace ov {
+namespace test {
+using adapoolParams = std::tuple<
+        std::vector<InputShape>,            // feature map shape
+        std::vector<int>,                   // pooled spatial shape
+        std::string,                        // pooling mode
+        ov::element::Type,                  // net type
```
Comment (vurusovs) [inline src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/adaptive_pooling.hpp:15]: @iefode @olpipi the test has only one "model" type, input and output types are implicitly "undefined". Is it needed to add types explicitly as it's done in other tests (https://github.com/openvinotoolkit/openvino/pull/19597/files#diff-cc3700c4244ea84f102cd6ff2a47b78daf6054fb35d761131c0a8bf50594cc37R13-R15)?

Comment (olpipi) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/adaptive_pooling.hpp:15]: I cannot answer what option is correct. @iefode @ilyachur, Could you please help?

Comment (iefode) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/adaptive_pooling.hpp:15]: @olpipi NetType is absolutelu extra as I can see, we can define only input types as my opinion

Comment (olpipi) [reply src/tests/functional/shared_test_classes/include/shared_test_classes/single_op/adaptive_pooling.hpp:15]: Changed to usage inType and outType

File: src/tests/functional/shared_test_classes/src/single_op/adaptive_pooling.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    std::vector<InputShape> shapes;
+    std::vector<int> pooled_spatial_shape;
+    std::string pooling_mode;
+    ov::element::Type model_type;
+    std::tie(shapes, pooled_spatial_shape, pooling_mode, model_type, targetDevice) = this->GetParam();
+    init_input_shapes(shapes);
+
+    ov::ParameterVector params{std::make_shared<ov::op::v0::Parameter>(model_type, inputDynamicShapes.front())};
+
+    ngraph::Shape pooled_shape = {pooled_spatial_shape.size()};
```
Comment (ilyachur) [inline src/tests/functional/shared_test_classes/src/single_op/adaptive_pooling.cpp:48]: Please remove ngraph namespace

File: src/tests/functional/shared_test_classes/src/single_op/batch_to_space.cpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    std::vector<InputShape> shapes;
+    std::vector<int64_t> block_shape, crops_begin, crops_end;
+    ov::element::Type model_type;
+    std::tie(block_shape, crops_begin, crops_end, shapes, model_type, targetDevice) = this->GetParam();
+    init_input_shapes(shapes);
+
+    ov::ParameterVector params{std::make_shared<ov::op::v0::Parameter>(model_type, inputDynamicShapes.front())};
+
+    auto const_shape = ov::Shape{inputDynamicShapes.front().size()};
+    auto block_shape_node = std::make_shared<ov::op::v0::Constant>(ngraph::element::i64, const_shape, block_shape.data());
```
Comment (ilyachur) [inline src/tests/functional/shared_test_classes/src/single_op/batch_to_space.cpp:44]: ngraph namespace ...

File: src/tests/functional/shared_test_classes/src/single_op/adaptive_pooling.cpp (hunk 2 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/single_op/adaptive_pooling.cpp]: fixed

File: src/tests/functional/shared_test_classes/src/single_op/batch_to_space.cpp (hunk 3 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/single_op/batch_to_space.cpp]: fixed

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp [hunk 4]
```diff
@@ -963,6 +963,50 @@ ov::runtime::Tensor generate(const
     return tensor;
 }
 
+namespace comparison {
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:966]: `comparation`

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp [hunk 5]
```diff
@@ -963,6 +963,50 @@ ov::runtime::Tensor generate(const
...
+        } else if (i % 7 == 3) {
+            data_ptr[i] = std::numeric_limits<double>::quiet_NaN();
+        } else if (i % 7 == 5) {
+            data_ptr[i] = -std::numeric_limits<double>::quiet_NaN();
+        } else {
+            data_ptr[i] = start + static_cast<float>(random.Generate(range));
+        }
+    }
+}
+ov::runtime::Tensor generate(const
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:990]: should be out of `comparation` namespace

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp [hunk 6]
```diff
@@ -963,6 +963,50 @@ ov::runtime::Tensor generate(const
...
+                             std::shared_ptr<ov::op::v10::IsFinite>& node,
+                             size_t port,
+                             const ov::element::Type& elemType,
+                             const ov::Shape& targetShape) {
+    ov::Tensor tensor(elemType, targetShape);
+    fill_tensor(tensor);
+    return tensor;
+}
+ov::runtime::Tensor generate(const
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:999]: should be out of `comparation` namespace

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp [hunk 7]
```diff
@@ -963,6 +963,50 @@ ov::runtime::Tensor generate(const
...
+        }
+    }
+}
+ov::runtime::Tensor generate(const
+                             std::shared_ptr<ov::op::v10::IsFinite>& node,
+                             size_t port,
+                             const ov::element::Type& elemType,
+                             const ov::Shape& targetShape) {
+    ov::Tensor tensor(elemType, targetShape);
+    fill_tensor(tensor);
```
Comment (iefode) [inline src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:996]: `comparation::fill_tensor(tensor)`

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp (hunk 4 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:966]: done

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp (hunk 6 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp:1001]: done

File: src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp (hunk 7 above)
Comment (olpipi) [reply src/tests/functional/shared_test_classes/src/base/utils/generate_inputs.cpp]: done

**Review by vurusovs — APPROVED** [review]

**Pull request merged**